### PR TITLE
Add additional LLM span attributes

### PR DESCRIPTION
**Pull request by karthikscale3**
Fixes #

## Changes

Please provide a brief description of the changes here.

Adding additional attributes to LLM span attributes.

Note: if the PR is touching an area that is not listed in the [existing areas](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/README.md), or the area does not have sufficient [domain experts coverage](https://github.com/open-telemetry/semantic-conventions/blob/main/.github/CODEOWNERS), the PR might be tagged as [experts needed](https://github.com/open-telemetry/semantic-conventions/labels/experts%20needed) and move slowly until experts are identified.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: model/registry/gen-ai.yaml [hunk 1]
```diff
@@ -45,6 +45,30 @@ groups:
         brief: The top_p sampling setting for the LLM request.
         examples: [1.0]
         tag: llm-generic-request
+      - id: request.top_k
+        stability: experimental
+        type: double
+        brief: The top_k sampling setting for the LLM request.
+        examples: [1.0]
+        tag: llm-generic-request
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:53]: not needed (please remove from other attributes introduced in this PR)
```suggestion
```

File: model/registry/gen-ai.yaml [hunk 2]
```diff
@@ -45,6 +45,30 @@ groups:
...
+      - id: request.stop
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:54]:
```suggestion
      - id: request.stop_sequence
```

or 

```suggestion
      - id: request.stop_sequences
```

File: model/registry/gen-ai.yaml [hunk 3]
```diff
@@ -45,6 +45,30 @@ groups:
...
+        stability: experimental
+        type: double
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:56]:
```suggestion
        type: string[]
```

File: model/registry/gen-ai.yaml [hunk 4]
```diff
@@ -45,6 +45,30 @@ groups:
...
+        brief: The stop sequences to provide.
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:57]:
```suggestion
        brief: The stop sequences provided in the request.
```

File: .chloggen/839.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+change_type: enhancement
+component: gen-ai
+note: Adding top_k and stop to the list of required attributes for the gen_ai.request.model event.
```
Comment (lmolkova) [inline .chloggen/839.yaml:3]:
```suggestion
note: Adding `gen_ai.request.top_k`, `gen_ai.request.presence_penalty`,  `gen_ai.request.frequency_penalty` and `gen_ai.request.stop_sequences` attributes.
```

**Comment by karthikscale3** [comment]
thanks @lmolkova . I have addressed all your review comments.

File: model/registry/gen-ai.yaml (hunk 1 above)
Comment (karthikscale3) [reply model/registry/gen-ai.yaml]: thanks! done

**Review by lmolkova — APPROVED** [review]
yaml changes look good, but please make sure to regenerate markdown tables

File: model/registry/gen-ai.yaml [hunk 6]
```diff
@@ -45,6 +45,26 @@ groups:
...
+      - id: request.frequency_penalty
+        stability: experimental
+        type: double
+        brief: The frequency penalty to provide.
+        examples: ['0.1']
+      - id: request.presence_penalty
+        stability: experimental
+        type: double
+        brief: The presence penalty to provide.
+        examples: ['0.1']
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:67]:
```suggestion
        examples: [0.1]
```

File: model/registry/gen-ai.yaml [hunk 7]
```diff
@@ -45,6 +45,26 @@ groups:
...
+      - id: request.stop_sequences
+        stability: experimental
+        type: string[]
+        brief: The stop sequences provided in the request.
+        examples: ['\n']
+      - id: request.frequency_penalty
+        stability: experimental
+        type: double
+        brief: The frequency penalty to provide.
+        examples: ['0.1']
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:62]:
```suggestion
        examples: [0.1]
```

File: docs/attributes-registry/gen-ai.md [hunk 8]
```diff
@@ -19,6 +19,10 @@ This document defines the attributes used to describe telemetry in the context o
 | `gen_ai.request.model`           | string   | The name of the LLM a request is being made to.                                                  | `gpt-4`                                                                 | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `gen_ai.request.temperature`     | double   | The temperature setting for the LLM request.                                                     | `0.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `gen_ai.request.top_p`           | double   | The top_p sampling setting for the LLM request.                                                  | `1.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `gen_ai.request.stop`           | string   | Up to 4 sequences where the API will stop generating further tokens.                             | `1.0`                                                                   | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```
Comment (nirga) [inline docs/attributes-registry/gen-ai.md:22]: There's a mismatch in the name between this and the yaml. You should generate this automatically

Comment (nirga) [inline docs/attributes-registry/gen-ai.md:22]: The comment is not clear to me. First, it says "4", but for example [cohere](https://docs.cohere.com/reference/chat) allows 5 sequences. Second, the example isn't right - it's a sequence so it should be a string, not a number. Third, type should be an array of strings IMO

Comment (karthikscale3) [reply docs/attributes-registry/gen-ai.md:22]: Good catch! Thanks @nirga, I will fix and push these up

Comment (karthikscale3) [reply docs/attributes-registry/gen-ai.md:22]: Got it! Will generate and push this back up

**Review by drewby — COMMENTED** [review]
@karthikscale3, can you resolve the conversations (including two changes from Liudmila and we can approve and merge)

File: model/registry/gen-ai.yaml [hunk 9]
```diff
@@ -45,6 +45,26 @@ groups:
...
+        stability: experimental
+        type: double
+        brief: The top_k sampling setting for the LLM request.
+        examples: [1.0]
+      - id: request.stop_sequences
+        stability: experimental
+        type: string[]
+        brief: The stop sequences provided in the request.
+        examples: ['\n']
+      - id: request.frequency_penalty
```
Comment (joaopgrassi) [inline model/registry/gen-ai.yaml:58]: Would this be a candidate for future attributes under `frequency`? Also the one below with `presence`? Should we reserve these namespaces already?

File: model/trace/gen-ai.yaml [hunk 10]
```diff
@@ -23,6 +23,14 @@ groups:
         requirement_level: recommended
       - ref: gen_ai.request.top_p
         requirement_level: recommended
+      - ref: gen_ai.request.top_k
+        requirement_level: recommended
+      - ref: gen_ai.request.stop
+        requirement_level: recommended
+      - ref: gen_ai.request.frequncy_penalty
```
Comment (marcalff) [inline model/trace/gen-ai.yaml:30]: Typo, which breaks CI.

**Comment by karthikscale3** [comment]
> @karthikscale3, can you resolve the conversations (including two changes from Liudmila and we can approve and merge)

Apologies for the delay. Updated the PR
- incorporated feedbacks and resolved comments
- regenerated markdowns

File: model/registry/gen-ai.yaml [hunk 11]
```diff
@@ -54,6 +54,29 @@ groups:
...
+        type: double
+        brief: The frequency penalty setting for the LLM request.
+        examples: [0.1]
+        tag: llm-generic-request
+      - id: request.presence_penalty
+        stability: experimental
+        type: double
+        brief: The presence penalty setting for the LLM request.
+        examples: [0.1]
+        tag: llm-generic-request
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:79]:
```suggestion
```
(not needed )

File: model/registry/gen-ai.yaml [hunk 12]
```diff
@@ -54,6 +54,29 @@ groups:
...
+        stability: experimental
+        type: string[]
+        brief: List of sequences that the model will use to stop generating further tokens.
+        examples: ['forest', 'lived']
+      - id: request.frequency_penalty
+        stability: experimental
+        type: double
+        brief: The frequency penalty setting for the LLM request.
+        examples: [0.1]
+        tag: llm-generic-request
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:73]:
```suggestion
```

File: model/registry/gen-ai.yaml [hunk 13]
```diff
@@ -54,6 +54,29 @@ groups:
         type: double
         brief: The top_p sampling setting for the GenAI request.
         examples: [1.0]
+        tag: llm-generic-request
+      - id: request.top_k
+        stability: experimental
+        type: double
+        brief: The top_k sampling setting for the LLM request.
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:61]:
```suggestion
        brief: The top_k sampling setting for the GenAI request.
```

File: model/registry/gen-ai.yaml [hunk 14]
```diff
@@ -54,6 +54,29 @@ groups:
...
+        examples: [1.0]
+      - id: request.stop_sequences
+        stability: experimental
+        type: string[]
+        brief: List of sequences that the model will use to stop generating further tokens.
+        examples: ['forest', 'lived']
+      - id: request.frequency_penalty
+        stability: experimental
+        type: double
+        brief: The frequency penalty setting for the LLM request.
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:71]:
```suggestion
        brief: The frequency penalty setting for the GenAI request.
```

File: model/registry/gen-ai.yaml [hunk 15]
```diff
@@ -54,6 +54,29 @@ groups:
...
+        examples: [0.1]
+        tag: llm-generic-request
+      - id: request.presence_penalty
+        stability: experimental
+        type: double
+        brief: The presence penalty setting for the LLM request.
```
Comment (lmolkova) [inline model/registry/gen-ai.yaml:77]:
```suggestion
        brief: The presence penalty setting for the GenAI request.
```

**Review by lmolkova — APPROVED** [review]
Left some cosmetic comment, LGTM otherwise

**Comment by karthikscale3** [comment]
> Left some cosmetic comment, LGTM otherwise

Thanks @lmolkova! I have updated the PR again. Looking for one more approval and then we should be good to merge this one. cc @drewby

**Review by nirga — APPROVED** [review]
@lmolkova I think this is good to go

**Comment by nirga** [comment]
Oh wait @karthikscale3 there’s some failures in CI

**Comment by lmolkova** [comment]
@karthikscale3 I think you have to regenerate md (`make attribute-registry-generation table-generation` )

**Comment by karthikscale3** [comment]
> make attribute-registry-generation table-generation

Thank you! Regenerated and pushed it back up. Will monitor for any other failures

**Pull request merged**